runtime/near-vm-runner/src/prepare.rs

```diff
@@ -44,48 +37,37 @@ impl<'a> ContractModule<'a> {
             elements::External::Memory(entry),
         ));
 
-        self.module = Some(builder.build());
+        Self { module: builder.build(), config }
     }
 
     /// Ensures that module doesn't declare internal memories.
     ///
     /// In this runtime we only allow wasm module to import memory from the environment.
     /// Memory section contains declarations of internal linear memories, so if we find one
     /// we reject such a module.
-    fn ensure_no_internal_memory(&self) -> Result<(), Error> {
-        let module =
-            self.module.as_ref().expect("On entry to the function `module` can't be None; qed");
-        if module.memory_section().map_or(false, |ms| !ms.entries().is_empty()) {
-            return Err(Error::InternalMemoryDeclared);
+    fn ensure_no_internal_memory(self) -> Result<Self, PrepareError> {
+        if self.module.memory_section().map_or(false, |ms| !ms.entries().is_empty()) {
+            Err(PrepareError::InternalMemoryDeclared)
+        } else {
+            Ok(self)
         }
-        Ok(())
     }
 
-    fn inject_gas_metering(&mut self) -> Result<(), Error> {
+    fn inject_gas_metering(self) -> Result<Self, PrepareError> {
+        let Self { module, config } = self;
         // TODO(#194): Re-enable .with_forbidden_floats() once AssemblyScript is fixed.
-        let gas_rules = rules::Set::new(self.config.regular_op_cost, Default::default())
-            .with_grow_cost(self.config.grow_mem_cost);
-
-        let module =
-            self.module.take().expect("On entry to the function `module` can't be `None`; qed");
-
-        let contract_module = pwasm_utils::inject_gas_counter(module, &gas_rules)
-            .map_err(|_| Error::GasInstrumentation)?;
-
-        self.module = Some(contract_module);
-        Ok(())
+        let gas_rules = rules::Set::new(config.regular_op_cost, Default::default())
+            .with_grow_cost(config.grow_mem_cost);
```

Comment:
Should we try `.with_forbidden_floats()`?

Reply:
Done